antismash/common/secmet/features/feature.py

```diff
@@ -69,7 +69,7 @@ def get_sub_location_from_protein_coordinates(self, start: int, end: int) -> Fea
         if start >= end:
             raise ValueError("Protein start coordinate must be less than the end coordinate")
 
-        dna_start, dna_end = convert_protein_position_to_dna(start, end, self.location)
+        dna_start, dna_end = convert_protein_position_to_dna(start, end, self.location, self.get_codon_start())
```

Comment:
Can we move this subroutine to CDSFeature class then? It seems that all the calls to it are from objects of CDSFeature class, except for one Prepeptide call

Reply:
regarding the Prepeptide. Since we would be using as much as 3 parameters from query cds when creating the Prepeptide() object (query.location, query.get_name() and query.get_codon_start()), wouldn't it be better to just pass "query" as "parent_cds" or something and do the calling to those 3 within the Prepeptide() class? Or is it a very bad idea?

Reply:
oh, i can see now that with to_json / from_json serializations, this might become a problem

Reply:
forget the questions about changing what is being passed to Prepeptide() constructor, if you got them over the email. I understand that the presence of to_json/from_json and to_biopython/from_biopython makes the idea of passing the whole CDSFeature as a parameter to Prepeptide() impractical. I would still be glad if you can comment on theoretical move of get_sub_location_from_protein_coordinates() to the CDSFeature class. Thanks.